src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -919,23 +919,37 @@ private void writeTypeInfo(TDSWriter tdsWriter, int srcJdbcType, int srcScale, i
 
             case microsoft.sql.Types.MONEY:
             case microsoft.sql.Types.SMALLMONEY:
+                tdsWriter.writeByte(TDSType.MONEYN.byteValue()); // 0x6E
+                if (SSType.MONEY == destSSType)
+                    tdsWriter.writeByte((byte) 8);
+                else
+                    tdsWriter.writeByte((byte) 4);
+                break;
             case java.sql.Types.NUMERIC:
             case java.sql.Types.DECIMAL:
-                if (isBaseType && ((SSType.MONEY == destSSType) || (SSType.SMALLMONEY == destSSType))) {
-                    tdsWriter.writeByte(TDSType.MONEYN.byteValue()); // 0x6E
-                    if (SSType.MONEY == destSSType)
-                        tdsWriter.writeByte((byte) 8);
-                    else
-                        tdsWriter.writeByte((byte) 4);
-                } else {
-                    if (java.sql.Types.DECIMAL == srcJdbcType)
-                        tdsWriter.writeByte(TDSType.DECIMALN.byteValue()); // 0x6A
-                    else
-                        tdsWriter.writeByte(TDSType.NUMERICN.byteValue()); // 0x6C
-                    tdsWriter.writeByte((byte) TDSWriter.BIGDECIMAL_MAX_LENGTH); // maximum length
-                    tdsWriter.writeByte((byte) srcPrecision); // unsigned byte
-                    tdsWriter.writeByte((byte) srcScale); // unsigned byte
+                /*
+                 * SQL Server allows the insertion of decimal and numeric into a money (and smallmoney) column,
+                 * but Azure DW only accepts money types for money column.
+                 * To make the code compatible against both SQL Server and Azure DW, always send decimal and
+                 * numeric as money/smallmoney if the destination column is money/smallmoney
+                 * and the source is decimal/numeric.
+                 */
+                if (destSSType == SSType.MONEY) {
+                    tdsWriter.writeByte(TDSType.MONEYN.byteValue());
+                    tdsWriter.writeByte((byte) 8);
+                    break;
+                } else if (destSSType == SSType.SMALLMONEY) {
+                    tdsWriter.writeByte(TDSType.MONEYN.byteValue());
+                    tdsWriter.writeByte((byte) 4);
+                    break;
                 }
+                if (java.sql.Types.DECIMAL == srcJdbcType)
+                    tdsWriter.writeByte(TDSType.DECIMALN.byteValue()); // 0x6A
+                else
+                    tdsWriter.writeByte(TDSType.NUMERICN.byteValue()); // 0x6C
```

Comment:
```byte byteType = (java.sql.Types.DECIMAL == srcJdbcType) ? TDSType.DECIMALN.byteValue() : TDSType.NUMERICN.byteValue();
tdsWriter.writeByte(byteType);```

Change if block to above to shorten the code and make it more clear.

Reply:
done